dexter/src/main/java/com/karumi/dexter/listener/threaddecorator/ThreadSpec.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright (C) 2015 Karumi.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.karumi.dexter.listener.threaddecorator;
+
+/**
+ * A thread specification to execute passed runnable objects in a certain thread
+ */
+public interface ThreadSpec {
```

Comment:
IMHO the only reason we have to use this abstraction is to avoid testing problems related to the Handler mocked implementation and reduce the library coupling. But I'd change the name to ``Thread`` instead of ``ThreadSpec``.